ai/unet/Model Training.py

```
@@ -0,0 +1,528 @@
+# ═══════════════════════════════════════════════════════════════════════════
+# TRAINING LOOPS
+# ═══════════════════════════════════════════════════════════════════════════
+
+def train_epoch(model, loader, criterion, optimizer, scaler, device):
+    model.train()
+    tot_loss = tot_disc = tot_cup = 0.0
+    use_amp  = (device == 'cuda')
+
+    pbar = tqdm(loader, desc="[TRAIN]", leave=False)
+    for images, masks in pbar:
+        images, masks = images.to(device), masks.to(device)
```

Ruff (0.15.7)

[warning] `for` loop variable `images` overwritten by assignment target

(PLW2901)